spec/apps/partial-rerouting/partial-rerouting.spec.js

```diff
@@ -0,0 +1,130 @@
+import * as singleSpa from "single-spa";
+
+describe("partial rerouting", () => {
+  function delay(ms) {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+  }
+
+  beforeAll(() => singleSpa.start());
+
+  beforeEach(() => {
+    location.hash = "#";
+  });
+
+  it("shoulnd't bootstrap and mount app if the route has been changed while app was loading, but if the user navigates back it should be bootstrapped and mounted successfully", async () => {
+    const activeHash = "#partial-rerouting";
+
+    let bootstrapped = false;
+    let mounted = false;
+    let unmounted = false;
+
+    const app = {
+      async bootstrap() {
+        bootstrapped = true;
+      },
+      async mount() {
+        mounted = true;
+      },
+      async unmount() {
+        unmounted = true;
+      },
+    };
+
+    singleSpa.registerApplication(
+      "partial-rerouting",
+      () => delay(1000).then(() => app),
+      (location) => location.hash === activeHash
+    );
+
+    location.hash = activeHash;
+
+    const promise = singleSpa.triggerAppChange();
+    await delay(100);
```

Comment:
To avoid waiting in real time for tests, you can use jest's fake timers - https://jestjs.io/docs/en/timer-mocks

100ms isn't a lot of time, but if done in every test it adds up. The rest of the tests are consistent about using fake timers, but I think it would be nice to add here if you are up to it.

Reply:
Okay, no problem.